demos/smart_framing_demo/cpp_gapi/include/custom_kernels.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <opencv2/gapi/gkernel.hpp>
+#include <opencv2/gapi/infer/ie.hpp>
+
+#include <inference_engine.hpp>
+
+#include <utils/slog.hpp>
+
+namespace custom {
+
+struct DetectedObject : public cv::Rect2f
+{
+    unsigned int labelID;
+    std::string label;
+    float confidence;
+};
```

Comment:
I doubt is subclassing from Rect2f is a good idea.. Not sure if it is bad, too.

The reason we _usually_ don't use structures like this in pipeline and instead produce multiple arrays of different types is that it is easier to process those arrays with other functions like `infer-list` without any extra custom code.